### ISSUE TITLE
[digitalstrom] StringIndexOutOfBoundsException

### ISSUE THREAD

**Issue by alexf2015** [openhab/openhab-addons#9048]
```
java.lang.StringIndexOutOfBoundsException: String index out of range: 2
        at java.lang.String.substring(String.java:1963) ~[?:1.8.0_211]
        at org.openhab.binding.digitalstrom.internal.discovery.DiscoveryServiceManager.onDeviceRemoved(DiscoveryServiceManager.java:179) ~[?:?]
        at org.openhab.binding.digitalstrom.internal.lib.manager.impl.DeviceStatusManagerImpl.checkDeviceConfig(DeviceStatusManagerImpl.java:651) ~[?:?]
        at org.openhab.binding.digitalstrom.internal.lib.manager.impl.DeviceStatusManagerImpl.access$8(DeviceStatusManagerImpl.java:620) ~[?:?]
        at org.openhab.binding.digitalstrom.internal.lib.manager.impl.DeviceStatusManagerImpl$PollingRunnable.run(DeviceStatusManagerImpl.java:283) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_211]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_211]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_211]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_211]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_211]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_211]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_211]
```
This exception occoured during update of my digitalstrom server. Thus, it will not be easy to reproduce but it should be quite easy to fix.

**Comment by gaetancollaud** [comment]
I had the exact same issue but it was not during an update. It just happens.

```
2020-11-19 18:55:27.494 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.StringIndexOutOfBoundsException: String index out of range: 2
	at java.lang.String.substring(String.java:1963) ~[?:1.8.0_265]
	at org.openhab.binding.digitalstrom.internal.discovery.DiscoveryServiceManager.onDeviceAdded(DiscoveryServiceManager.java:198) ~[?:?]
	at org.openhab.binding.digitalstrom.internal.lib.manager.impl.DeviceStatusManagerImpl$PollingRunnable.run(DeviceStatusManagerImpl.java:346) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_265]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
2020-11-19 19:01:16.027 [WARN ] [mmon.WrappedScheduledExecutorService] - Scheduled runnable ended with an exception: 
java.lang.StringIndexOutOfBoundsException: String index out of range: 2
	at java.lang.String.substring(String.java:1963) ~[?:1.8.0_265]
	at org.openhab.binding.digitalstrom.internal.discovery.DiscoveryServiceManager.onDeviceAdded(DiscoveryServiceManager.java:198) ~[?:?]
	at org.openhab.binding.digitalstrom.internal.lib.manager.impl.DeviceStatusManagerImpl$PollingRunnable.run(DeviceStatusManagerImpl.java:346) ~[?:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_265]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308) ~[?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180) ~[?:1.8.0_265]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294) ~[?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_265]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_265]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_265]
```

At that exact same time I had issue with my digitalstrom system. The system was unresponsive, but worked again one hour later for some reason...  This happend two times while we used the hair dryer (maybe some disturbance on the power network...). I know it's not related to openhab but the fact that digital strom crash somehow makes this addon crash too (and unable to recover).

If I have time I will come with a pull request to secure this part of the code.

**Comment by 9037568** [comment]
Fixed by #9194
